base/doc/clsguide.tex

```diff
@@ -977,6 +977,30 @@ \subsection{Passing options around}
 handler, but this handler is not used for |landscape| because that option is
 explicitly declared.
 
+\subsection{Testing dates of other code}
+
+\begin{decl}
+  |\IfPackageAtLeast(TF)| \arg{package} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfClassAtLeast(TF)| \arg{class} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfFileAtLeast(TF)| \arg{file} \arg{date} \arg{true code} \arg{false code}
+\end{decl}
+There are occasions when you need to test if one package if another package has
+a particular version (assuming it has been loaded). This can be done using the
+commands |\IfPackageAtLeastTF|, etc. Separate commands available to test
+packages, classes and general files: in all cases, they rely on the existence
+of a |\Provides...| line in the appropriate file being queried. Three versions
+of each command are available, ending |T|, |F| and |TF|, which indicate the
+outcome branches required. Thus for example |\IfPakcageAtLeastT| does not need
```

Comment:
Maybe:
. . . which specify the branches provided."

Reply:
I was trying to get at that you need particular arguments - not sure if 'provided' covers that.

Reply:
OK, so say something about the arguments.
Maybe a reference to the very many other similar "T, F or TF variant" cases would be best.